### ISSUE TITLE
Bug: Validation functions that take an array as a parameter only receiving the first element of the array in the proper parameter

### ISSUE THREAD

**Issue by ThatOneBro** [honojs/validator#6]
Hi @yusukebe , I just want to say I really appreciate your work on Hono. Hono is great and fun to work with! 😄 

I just started trying out this package and I noticed that validation was failing on all parameters using the `v.isIn` validator. `v.isIn` accepts an array and checks if the input provided is included in the array. I think because of the parameter spreading done like `[Function, ...Param[]]` it is not possible for the entire array to be passed to the validator as it is spread across the params rather than passed as the array to the function. 

**Example 1**:
```js
app.post("/*",
  validation((v) => ({
    json: {
      value: [v.required, [v.isIn, ["valid", "also_valid"]],
    },
  })),
);
```
When `value` is `valid` it passes validation.
When `value` is `also_valid` it doesn't.

I was able to get the desired functionality with a custom validator that uses spread syntax in the function signature.
   
**Example 2**:
```ts
const customIsIn = (value: string, ...arr: string[]) => {
  return arr.includes(value);
};

app.post("/*",
  validation((v) => ({
    json: {
      value: [v.required, [customIsIn, ["valid", "also_valid"]],
    },
  })),
);
```
`Value` passes when it is `valid` or `also_valid`.

This is obviously a work-around, though. I think conditional spreading is necessary but at the moment while writing this I can't think of the best way to handle this; however I wanted to make sure I noted the issue before I continue thinking about a solution.

I'll update this issue once I think about it some more. Again thanks for all your hardwork @yusukebe 😄 I hope I can help to contribute to the Hono project as I work with it.

**Comment by yusukebe** [comment]
Hi @ThatOneBro !

Thanks a lot!
Your report is helpful because the validator middleware has just been created and has not been fully tested in practice.

This bug has been fixed in this PR #8 . The tests below are passed. It will work as expected.

https://github.com/honojs/validator/pull/8/files#diff-ec61822d929089017a892b4b047229377c9241f16b43252ab851ae39131cdf40R187

I'll release the next version, including this bug fix, soon!

And, thank you for giving me glad words. Contribution welcome!

**Comment by yusukebe** [comment]
Hey! I've released v0.1.1, try it!

**Comment by ThatOneBro** [comment]
> Hey! I've released v0.1.1, try it!

Wow that was fast! It's working great now. Thanks for responding so quickly to my issues! I was able to remove my hacks now 😄

**Comment by yusukebe** [comment]
Hi!

> I will try to work on some issues soon if I get a chance, are there any issues around the Hono project which should be prioritized?

Ah... Documentation? 🙂
